graphs/graph_modules/data_fetcher.py

```
@@ -0,0 +1,365 @@
+                        data[graph_type] = result['response']['data']
+                    else:
+                        data[graph_type] = result
+                    logging.debug("Successfully fetched data for %s", graph_type)
+                else:
+                    logging.warning(f"No data returned for {graph_type}")
+
+            return data
+                
+        except Exception as e:
+            logging.error(f"Error fetching graph data: {str(e)}")
+            raise DataFetcherError(f"Failed to fetch graph data: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)